src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -28,9 +28,18 @@ trait TypeAssigner {
     }
   }
 
-  def avoid(tp: Type, syms: => List[Symbol])(implicit ctx: Context): Type = {
+  /** An upper approximation of the given type `tp` that does not refer to any symbol in `symsToAvoid`.
```

Comment:
Agreed, but I don't have time to do that. Way too many other things on my plate :-(